api/v3/Case/Getdetails.php

```diff
@@ -235,6 +240,39 @@ function civicrm_api3_case_getdetails($params) {
   return $resultMetadata + $result;
 }
 
+/**
+ * Calculates the number of activities for the given category
+ *
+ * @param {String} $category
+ * @param {Array} $ids
+ * @param {String} $statusTypeIds
+ * @param {String} $statusType
+ * @param {Boolean} $isOverdue
+ * @param {Array} $result
+ */
+function calculate_activities_for_category($category, $ids, $statusTypeIds, $statusType, $isOverdue, &$result) {
+  $isOverdueCondition = $isOverdue ? "AND a.activity_date_time < NOW()" : "";
+  $categoryCondition = empty($category) ? "IS NULL" : "LIKE '%$category%'";
+
+  $query = "SELECT COUNT(a.id) as count, ca.case_id
+  FROM civicrm_activity a, civicrm_case_activity ca
+  WHERE ca.activity_id = a.id AND a.is_current_revision = 1 AND a.is_test = 0 AND ca.case_id IN (" . implode(',', $ids) . ")
+  AND a.activity_type_id IN (SELECT value FROM civicrm_option_value WHERE grouping "
+  . $categoryCondition ." AND option_group_id = (SELECT id FROM civicrm_option_group WHERE name = 'activity_type'))
+  ". $isOverdueCondition ."
+  AND is_current_revision = 1
+  AND is_deleted = 0
+  AND a.status_id IN ($statusTypeIds)
+  GROUP BY ca.case_id";
+
+  $dao = CRM_Core_DAO::executeQuery($query);
+
+  while ($dao->fetch()) {
+    $categoryName = empty($category) ? 'none' : $category;
+    $statusTypeName = $isOverdue ? "overdue" : $statusType;
+    $result['values'][$dao->case_id]['category_count'][$categoryName][$statusTypeName] = (int) $dao->count;
+  }
+}
```

Comment:
I'm ok with the solution, but I'm worried that we are fetching all this data even for cases that are not going to display the count (like the ones in the list table). Can you confirm this for me?

Reply:
As discussed, this is fine.